code.go

```diff
@@ -40,10 +39,11 @@ var SourceCodeTypes = map[string]bool{
 }
 
 // parseCode parses a code address directive.
-func parseCode(addr []byte, file []byte) []byte {
+func (p *parser) parseCode(addr []byte, file []byte) []byte {
 	bytes.TrimSpace(addr)
 
-	textBytes, err := ioutil.ReadFile(string(file))
+	fullname := absname(p.cwd, string(file))
```

Comment:
absname and ReadFile can just take []byte's, as that is the thing we are dealing with here anyway.